app/main/plugins/hain-plugin-filesearch/shared-util.js

```diff
@@ -14,10 +14,21 @@ function injectEnvVariable(dirPath) {
   }
 
   // Inject Environment Variables
-  for (const envVar in process.env) {
-    const value = process.env[envVar];
-    _path = _path.replace(`\${${envVar}}`, value);
+  let tokens = dirPath.match(/\${.*?}/g);
```

Comment:
Are you sure this is right?  Typically `{` and `}` designate min/max values for the previous character (in this case the escaped $).  I think you're wanting `(` and `)`.

Reply:
The tokens are literally ${SystemHost} out of a string like ${SystemHost}/my/path - this regex seemed to work in my own testing but feel free to suggest a different regex for these matches.

Reply:
Perhaps 
```
\$\{(.*?)\}
```
?

This should allow the removal of the subsequent slice operation.